service/matching/matching_engine.go

```diff
@@ -837,6 +837,118 @@ func (e *matchingEngineImpl) listTaskQueuePartitions(request *matchingservice.Li
 	return partitionHostInfo, nil
 }
 
+func (e *matchingEngineImpl) UpdateWorkerVersioningRules(
+	ctx context.Context,
+	req *workflowservice.UpdateWorkerVersioningRulesRequest,
+) (*workflowservice.UpdateWorkerVersioningRulesResponse, error) {
+	ns, err := e.namespaceRegistry.GetNamespace(namespace.Name(req.GetNamespace()))
+	if err != nil {
+		return nil, err
+	}
+
+	taskQueue, err := newTaskQueueID(ns.ID(), req.GetTaskQueue(), enumspb.TASK_QUEUE_TYPE_WORKFLOW)
+	if err != nil {
+		return nil, err
+	}
+
+	tqMgr, err := e.getTaskQueuePartitionManager(ctx, taskQueue, normalStickyInfo, true)
+	if err != nil {
+		return nil, err
+	}
+
+	// todo carly: do we need to update updateOptions.TaskQueueLimitPerBuildId or updateOptions.KnownVersion like in lines 1009-1015?
```

Comment:
@dnr do you know why they did this before the `UpdateUserData` fn in `UpdateWorkerBuildIdCompatibility`? and do you know whether it makes sense to do anything with these two update options in the new versioning system? It seems like maybe `ConflictToken` has replaced `KnownVersion`, but I'm not sure. Here's the snippet from the other updatefn https://github.com/temporalio/temporal/blob/main/service/matching/matching_engine.go#L919-L925